src/main/scala/com/newlibertie/pollster/impl/Poll.scala

```diff
@@ -28,16 +31,49 @@ object Poll {
     }
     new Poll(pollParameters, cryptographicParameters)
   }
-
-  def write(poll: Poll) = {
-    DataAdapter.createPoll(poll)
-  }
-
   def read(id:String) = {
-    DataAdapter.getPoll(id)
-
+    DataAdapter.getPoll(id) match {
+      case m:mutable.Map[String, Any] =>
```

Comment:
what happens when data is bad and some exception is throws.  can we emit a statement there and ensure flow is correct - even to the extent of checking that behavior in test case